### PR TITLE
fix: AbortController and AbortSignal.timeout Polyfill

### PR DESCRIPTION
**Pull request by chef-ryan**
- Introduced a polyfill for AbortController and AbortSignal.timeout, enabling support for aborting operations in environments that do not natively support these features.
- This addition allows developers to use AbortController to manage request cancellations more effectively, improving error handling and resource management.

---

## PR-Codex overview
This PR introduces the `abortcontroller-polyfill` to enhance compatibility with the `AbortSignal` API in the application. It also updates package dependencies and modifies the `package.json` to include the new polyfill.

### Detailed summary
- Added `abortcontroller-polyfill` import in `apps/web/src/pages/_app.tsx`.
- Implemented a timeout function in `apps/web/src/utils/abortcontroller-polyfill.ts`.
- Updated `pnpm-lock.yaml` to include `abortcontroller-polyfill` version 1.7.5.
- Updated `@noble/curves` from 1.4.0 to 1.6.0 in `pnpm-lock.yaml`.
- Updated `ws` from 8.17.1 to 8.18.0 in `pnpm-lock.yaml`.
- Added `@kyberswap/pancake-liquidity-widgets` version 0.1.1 to `apps/web/package.json`.
- Removed `@kyberswap/pancake-liquidity-widgets` from previous dependencies in `apps/web/package.json`.
- Added `abortcontroller-polyfill` version 1.7.5 to `apps/web/package.json`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -4,18 +4,6 @@ settings:
...
-    next:
-      specifier: 14.2.14
-      version: 14.2.14
-    viem:
-      specifier: ^2.21.22
-      version: 2.21.22
-    wagmi:
-      specifier: ^2.12.17
-      version: 2.12.17
-
```
Comment (chefjackson) [inline pnpm-lock.yaml:18]: why catalogs are removed?

Comment (chefjackson) [reply pnpm-lock.yaml:18]: Please make sure we're generating the lockfile using the correct pnpm version specified in the pacakge.json

File: apps/web/src/index.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { useAccount } from 'wagmi'
 import ListsUpdater from './state/lists/updater'
 import MulticallUpdater from './state/multicall/updater'
 import TransactionUpdater from './state/transactions/updater'
+import './utils/abortcontroller-polyfill'
```
Comment (chefjackson) [inline apps/web/src/index.tsx:10]: Shall we add to `_app.tsx` instead?

File: pnpm-lock.yaml (hunk 1 above)
Comment (chef-ryan) [reply pnpm-lock.yaml]: Thx. Fixed after synced local pnpm version.

File: apps/web/src/index.tsx (hunk 2 above)
Comment (chef-ryan) [reply apps/web/src/index.tsx]: Done.

**Review by chefjackson — APPROVED** [review]

**Pull request merged**